### PR TITLE
Use `region` field as comma-delimited node selector for k8s

### PR DESCRIPTION
**Pull request by sjawhar**
* Adds the ability to use the `region` field of the task in the Terraform file to pass node selector labels to the job.
* Expected format is comma-delimited `key=value` pairs
   * e.g.  `region: foo=bar,goo=baz,...`
   * Hacky?

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
> _Hacky?_

Yes, _cough, cough_, yes: https://github.com/iterative/terraform-provider-iterative/issues/412#issuecomment-1159729184. Nevertheless, sounds useful to me.

**Comment by dacbd** [comment]
willing to add some documentation snippets?

**Comment by sjawhar** [comment]
> Yes, _cough, cough_, yes: [#412 (comment)](https://github.com/iterative/terraform-provider-iterative/issues/412#issuecomment-1159729184). Nevertheless, sounds useful to me.

I got the impression from the convo on my last PR that we didn't want to add cloud-specific options, but I'm happy to decline this PR and do it that way if preferred :smile:

**Comment by 0x2b3bfa0** [comment]
We have some mixed opinions on the topic, but this proposal sounds good to me. It's a stretch of the “region” concept, but at least it adds some support for `k8s` node selectors instead of outright ignoring the `region` field.

**Review by tasdomas — APPROVED** [review]

**Comment by sjawhar** [comment]
It looks like the smoke test failed because of "Quota Exceeded"

**Comment by omesser** [comment]
> It's a stretch of the “region” concept, but at least it adds some support for k8s node selectors instead of outright ignoring the region field.

Yes indeed :) 
I have not strong objection to merge this (even if it becomes a small tech debt),
but after merging this  - let's please discuss proper support for k8s specific configuration (it's not a cloud provider), cc @iterative/cml
I am not convinced we're doing anything other than delaying the inevitable and confusing users with mixing up of terminologies 🥲

File: docs/resources/task.md
Comment (0x2b3bfa0) [inline docs/resources/task.md:57]:
```suggestion
- `region` - (Optional) [Cloud region/zone](#cloud-region) to run the task on, or node selector labels for Kubernetes.
```

File: docs/resources/task.md
Comment (0x2b3bfa0) [inline docs/resources/task.md:260]:
```suggestion
For Kubernetes, the `region` attribute can be used to set the [`nodeSelector`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) field of the job specification. The value of `region` should be a comma-delimited list of `key=value` nodel label pairs.
```

**Review by 0x2b3bfa0 — COMMENTED** [review]
Some cosmetic documentation changes 💅🏼

**Comment by 0x2b3bfa0** [comment]
Also passwd the Kubernetes smoke test :shipit:

**Pull request merged**
